meta/fraction-metadata/src/main/java/org/wildfly/swarm/fractions/scanner/FilePresenceScanner.java

```diff
@@ -22,12 +22,13 @@
 import java.util.function.Consumer;
 
 import org.wildfly.swarm.spi.meta.FileDetector;
+import org.wildfly.swarm.spi.meta.FileSource;
 import org.wildfly.swarm.spi.meta.FractionDetector;
 
 /**
  * @author Heiko Braun
  */
-public class FilePresenceScanner implements Scanner<String> {
+public class FilePresenceScanner implements Scanner<FileSource> {
     @Override
     public String extension() {
         return XML;
```

Comment:
I think it's fine to have this kind of mechanism just in case someday file extension won't be XML. But maybe we could get rid of it as having the file name should be enough and extension is contained there too.